charts/ScatterPlot.tsx

```diff
@@ -89,9 +89,15 @@ export class ScatterPlot extends React.Component<{
                 return that.chart.baseFontSize
             },
             get colorables() {
-                return that.transform.colors.colorables.filter(c =>
-                    that.colorsInUse.includes(c.color)
-                )
+                return that.transform.colorScale.legendData
+                    .filter(bin => that.colorsInUse.includes(bin.color))
```

Comment:
A slight orthogonal bug with this in the explorer that if you click a region and then all of those countries are selected, because they are then assigned fixed colors and no country then is using the "region" color, that region will disappear from the legend. Not something to fix here, just wanted to mention it in case you had any solution ideas.

Reply:
Ah right, we would have to disable the automatic color assignment happening in the country selector. Will do it in this PR if I can find an easy way.

Reply:
Ok I just understood a bit better what you mean.

The current legend matches bins to "point groups" by the `group.color` property. I think that property should contain the original value rather than the derived color from the value.

So e.g. it should contain the tests per case (`6`) instead of the derived color from that (`#c32623`). Those values should be converted to colors at the very end, when rendering the components.

This is how the map legend works – it matches bins by the numeric value instead of the color value. But `ScatterColorLegend` is still looking at derived colors. 

I would like to change it but it includes changing the hover/focus behaviour or Scatter, StackedBar, SlopeChart, and we also don't yet have a good idea for what the legend should ideally look like, and how continuous legends could work.